plenum/server/consensus/checkpoint_service.py

```diff
@@ -368,58 +234,51 @@ def update_watermark_from_3pc(self):
         else:
             self._logger.info("try to update_watermark_from_3pc but last_ordered_3pc is None")
 
-    def _remove_stashed_checkpoints(self, till_3pc_key=None):
+    def _remove_received_checkpoints(self, till_3pc_key=None):
         """
-        Remove stashed received checkpoints up to `till_3pc_key` if provided,
-        otherwise remove all stashed received checkpoints
+        Remove received checkpoints up to `till_3pc_key` if provided,
+        otherwise remove all received checkpoints
         """
         if till_3pc_key is None:
-            self._stashed_recvd_checkpoints.clear()
-            self._logger.info('{} removing all stashed checkpoints'.format(self))
+            self._received_checkpoints.clear()
+            self._logger.info('{} removing all received checkpoints'.format(self))
             return
 
-        for view_no in list(self._stashed_recvd_checkpoints.keys()):
-
-            if view_no < till_3pc_key[0]:
-                self._logger.info('{} removing stashed checkpoints for view {}'.format(self, view_no))
-                del self._stashed_recvd_checkpoints[view_no]
+        def is_below_3pc_key(cp: CheckpointService.CheckpointKey, key: Tuple[int, int]):
```

Comment:
Can we use `compare_3PC_keys` instead, especially taking into account INDY-1336 and an upcoming logic of not resetting ppSeqNo?

Reply:
Now `compare_3PC_keys` is used inside `is_below_3pc_key`, so yes, it can be used